opm/simulators/flow/LogOutputHelper.cpp

```diff
@@ -306,34 +308,77 @@ fip(const Inplace& inplace,
     }
 }
 
+
 template<class Scalar>
 void LogOutputHelper<Scalar>::
-fipResv(const Inplace& inplace) const
+fipResv(const Inplace& inplace, const std::string& name) const
 {
     {
         std::unordered_map<Inplace::Phase, Scalar> current_values;
 
-        for (const auto& phase : Inplace::phases()) {
+        for (const auto& phase : Inplace::phases()) 
             current_values[phase] = inplace.get(phase);
-        }
+            
+        Scalar field_dyn_pv = 0.0;
+        
+        for (std::size_t reg = 1; reg <= inplace.max_region(name); ++reg)
+            field_dyn_pv = field_dyn_pv + inplace.get(name, Inplace::Phase::DynamicPoreVolume, reg);
+        
+        current_values[Inplace::Phase::DynamicPoreVolume] = field_dyn_pv;
+        
         this->fipUnitConvert_(current_values);
         this->outputResvFluidInPlace_(current_values, 0);
     }
-
-    for (std::size_t reg = 1; reg <= inplace.max_region("FIPNUM"); ++reg) {
+    
+    for (std::size_t reg = 1; reg <= inplace.max_region(); ++reg) {
```

Comment:
Same here.  If we're changing this anyway, I'd prefer we compute `inplace.max_region()` only once.